src/renderer/components/+namespaces/namespace-select.tsx

```diff
@@ -70,38 +71,61 @@ export class NamespaceSelect extends React.Component<Props> {
       <Select
         className={cssNames("NamespaceSelect", className)}
         menuClass="NamespaceSelectMenu"
-        formatOptionLabel={this.formatOptionLabel}
+        autoConvertOptions={false}
         options={this.options}
         {...selectProps}
       />
     );
   }
 }
 
+interface BasicNS {
+  label: React.ReactElement;
+  value: string;
+}
+
 @observer
 export class NamespaceSelectFilter extends React.Component {
+  async componentDidMount() {
+    if (!namespaceStore.isLoaded) {
+      await namespaceStore.loadAll();
+      namespaceStore.contextNs.clear();
+    }
+  }
+
+  @autobind()
+  onChange(value: BasicNS, actionMeta: ActionMeta<BasicNS>) {
+    switch (actionMeta.action) {
+    case "select-option":
+      namespaceStore.contextNs.add(actionMeta.option.value)
+      break
+    case "clear":
+      namespaceStore.contextNs.clear()
+      break
+    case "deselect-option":
+      namespaceStore.contextNs.delete(actionMeta.option.value)
+      break
+    }
+  }
+
   render() {
-    const { contextNs, hasContext, toggleContext } = namespaceStore;
-    let placeholder = <Trans>All namespaces</Trans>;
-    if (contextNs.length == 1) placeholder = <Trans>Namespace: {contextNs[0]}</Trans>
-    if (contextNs.length >= 2) placeholder = <Trans>Namespaces: {contextNs.join(", ")}</Trans>
     return (
-      <NamespaceSelect
-        placeholder={placeholder}
+      <ReactSelect
```

Comment:
Do we use `NamespaceSelect` component somewhere after this change?

Reply:
Yes. For instance "add-quota-dialog".